### ISSUE TITLE
Visually Debugging Generated Markup

### ISSUE THREAD

**Issue by SMLMRKHLMS** [styled-components/styled-components#174]
### Feature Request:

First of all, I want to start off by saying how much I love this project. I really appreciate the thoughtfulness and hard work that went in to this wonderful library. Thanks!

One issue I have encountered is that I am having a bit of difficulty visually debugging my generated markup. The classNames that are generated obviously help with name-spacing and scoping styles; however, they are visually difficult to sift through when searching for a specific element. This is what my generated markup looks like inside of developer tools.
```html
<div class="bQOuPN">
  <h3 class="hgzzbv">Collaborations</h3>
  <div class="dFPIIa">
    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt.</p>
  </div>
</div>
```
It would be helpful if I had the ability to prepend the hash with a specified `className` or `prefix` like so:
```html
<div class="container-bQOuPN">
  <h3 class="header-hgzzbv">Collaborations</h3>
  <div class="text-dFPIIa">
    <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt.</p>
  </div>
</div>
```
I am not sure if this already possible; however, based off my research I did not see a clear way of doing so. I know I have the ability to pass is a className like so:
```jsx
<Container className="container"></Container>
```
and then reference the className inside of the `styled-component` like this:
```js
const Container = styled.div`
  &.container {

  }
`
```
This, however, feels like a lot of unnecessary duplication. I was thinking an interesting api addition could be something like this:
```js
const Container = styled.div({ prefix })`

`
```
Basically, this would also be giving one the ability to pass props directly to the styled component upon creation.

Let me know what you think. I wanted to start a discussion of potential solutions more than anything else. I may be off-base and there may be a better workaround, but I thought this might be a nice addition to the current api. Thanks for your time!

**Comment by oscar-b** [comment]
It would be nice to turn this on only for dev.

**Comment by mxstbr** [comment]
There is an open PR that we'll merge soon that'll add support for this feature, closing in favor of discussing in #160!
